### PR TITLE
#1825 agent windows auth from config

### PR DESCRIPTION
**Pull request by andrisarkameru**

### CODE REVIEW

**Comment by z1c0** [comment]
Thanks for the PR @andrisarkameru:

Before we can merge it though, let's:
- [ ] verify the basic functionality of this PR.
- [ ] agree on a name for this new setting (see my comment above).
- [ ] get a proper description of this nice feature in `configuration.asciidoc`.
- [ ] add the docs team to review as soon as the documentation input is finished.

**Review by z1c0 — CHANGES_REQUESTED** [review]
@andrisarkameru can I ask you to rename all occurrences of the new setting to `UseWindowsCredentials`, please?

File: docs/configuration.asciidoc
Comment (andrisarkameru) [reply docs/configuration.asciidoc:622]: Since basicly all it does is set the dotnet HttlClientHandler.UseDefaultCredentials property, maybe we can reuse some of the microsoft documentation for that: https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.usedefaultcredentials?view=net-6.0 

So maybe something like: 
"Set this property to true when requests made by the apm-agent should, if requested by the server, be authenticated using the credentials of the currently logged on user. This is useful when using windows authentication on a proxy, that routes apm-agent requests to the apm-server."

File: docs/configuration.asciidoc
Comment (andrisarkameru) [reply docs/configuration.asciidoc:626]: Yeah, that makes sense. Somehow, when I first wrote it, I wanted to be more descriptive, but UseWindowsCredentials sounds better to me too, because it resembles the "under-the-hood" property of the dotnet HttpClientHandler "UseDefaultCredentials" better.

File: docs/configuration.asciidoc
Comment (z1c0) [reply docs/configuration.asciidoc:622]: Perfect, I like it! Let's go with this.

File: docs/configuration.asciidoc
Comment (andrisarkameru) [reply docs/configuration.asciidoc]: renamed all the variables/props dropping the Server prefix

File: docs/configuration.asciidoc
Comment (andrisarkameru) [reply docs/configuration.asciidoc:626]: added asciidoc description

**Comment by z1c0** [comment]
@elastic/obs-docs could have a quick look at the doc change, please?

File: src/Elastic.Apm/Config/AbstractConfigurationReader.cs
Comment (z1c0) [inline src/Elastic.Apm/Config/AbstractConfigurationReader.cs:279]: Sorry for the nit ... can I ask you to add a blank line here for consistency reasons.

File: src/Elastic.Apm/Config/AbstractConfigurationWithEnvFallbackReader.cs
Comment (z1c0) [inline src/Elastic.Apm/Config/AbstractConfigurationWithEnvFallbackReader.cs:135]: same

File: src/Elastic.Apm/Config/EnvironmentConfigurationReader.cs
Comment (z1c0) [inline src/Elastic.Apm/Config/EnvironmentConfigurationReader.cs:112]: same

**Review by z1c0 — APPROVED** [review]
LGTM (apart from tiny nits 😉).
Thanks again for this contribution @andrisarkameru!

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/configuration.asciidoc
Comment (bmorelli25) [inline docs/configuration.asciidoc:623]: One small nit in regards to how we refer to APM agents in the docs.

```suggestion
Set this property to true when requests made by the APM agent should, if requested by the server, be authenticated using the credentials of the currently logged on user.
```

File: docs/configuration.asciidoc
Comment (bmorelli25) [inline docs/configuration.asciidoc:1303]: It looks like config variables are mostly alphabetized in this document. Is there a reason that this is the spot you chose? Maybe we can move this down after `UseElasticTraceparentHeader` to be more consistent?

Comment (andrisarkameru) [reply docs/configuration.asciidoc:1303]: Originally  before z1c0 suggestion for rename it was named with Server prefix and I put it next to the server-cert and server-token properties, because I thought that it grouped nicely in that theme because it is similar setting of APM agent connection to APM server. (that was also the reason I put the server prefix for that property). So after rename I have no objection to position it wherever fits best.

Comment (bmorelli25) [reply docs/configuration.asciidoc:1303]: Gotcha 👍. I think I'd still vote for alphabetical, but I'll leave it up to @z1c0.

Thanks for this contribution!

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Review by bmorelli25 — APPROVED** [review]

File: docs/configuration.asciidoc
Comment (z1c0) [reply docs/configuration.asciidoc:1303]: Agreed @bmorelli25, let's keep things consistent and go for alphabetical.

**Review by gregkalapos — APPROVED** [review]
LGTM.

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/configuration.asciidoc
Comment (bmorelli25) [inline docs/configuration.asciidoc:620]: This should be `config-use-windows-credentials` to match the other config anchors and the link in All options summary.

```suggestion
[[config-use-windows-credentials]]
```

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/configuration.asciidoc
Comment (andrisarkameru) [reply docs/configuration.asciidoc]: moved description bellow `UseElasticTraceparentHeader `

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by z1c0** [comment]
PR merged  🎉.
Thanks again for this contribution @andrisarkameru!

**Pull request merged**
